### DISCUSSION TITLE
Installation error: Microsoft Visual C++ 14.0 is needed

### DISCUSSION THREAD

**Discussion by Yayoee** [Dineshkarthik/telegram_media_downloader#400] [Q&A]
I hardly know things about code, could someone help me please?
When I followed the steps to run `pip3 install -r requirements.txt`, the error occured like this:
```
      running build_ext
      building 'tgcrypto' extension
      error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
      [end of output]
```
I followed the hint, but it was the same error when I tried to install the requirements again. And I searched a lot on the internet for the MSVC problems but there were no good solutions.

**Comment by tiagodamian** [comment]
You must find MS Build Tools 2019 download page.
there is tutorials on the web.
then download microsfot package.
MS Visual STudio installer is going to appear,
and select c++ package 6,25GB will be installed.

**Reply by Sebalux** [reply]
Follow the link in the error message
Download the VS Build Tools
Install "Desktop development with c++" package from the start page
About 6,7GB later, everything is done. Rerun pip3 install -r requirements.txt and see if that helps!
